File: test/testing.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

// HelloPrinter struct
type HelloPrinter struct{}

// PrintHello method
func (hp HelloPrinter) PrintHello() {
	delay := rand.Intn(4) + 2 // Random between 2 and 5
	time.Sleep(time.Duration(delay) * time.Second)
	fmt.Println("Hello World!")
}

func main() {
	rand.Seed(time.Now().UnixNano())
	hp := HelloPrinter{}
	hp.PrintHello()
}

File: test/TypeScript/testing.ts
import { sleep } from "https://deno.land/x/sleep/mod.ts";

class HelloPrinter {
  static async printHello() {
    const delay = Math.floor(Math.random() * (5 - 2 + 1)) + 2; // Random between 2 and 5
    await sleep(delay);
    console.log("Hello World!");
  }
}

if (import.meta.main) {
  await HelloPrinter.printHello();
}
